### PR TITLE
BUGZ-322 add domain checkin time-spent logging

### PR DESCRIPTION
**Pull request by roxanneskelly**
BUGZ-322 introduced an issue with white orbs.
This PR contains the original changes for BUGZ-322 plus a fix that moves the DomainList timestamps to the packet header so it's in every packet.

https://highfidelity.atlassian.net/browse/BUGZ-338

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19635/HighFidelity-Beta-PR15626-7fdbbe5.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19635/HighFidelity-Quest-Beta-PR15626-7fdbbe5.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19636/HighFidelity-Beta-PR15626-cc3ce96.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19636/HighFidelity-Beta-Interface-PR15626-cc3ce96.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19636/HighFidelity-Beta-PR15626-cc3ce96.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19636/HighFidelity-Quest-Beta-PR15626-cc3ce96.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19636/HighFidelity-Beta-PR15626-cc3ce96.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19636/HighFidelity-Beta-Interface-PR15626-cc3ce96.exe

File: domain-server/src/DomainServer.cpp [hunk 1]
```diff
@@ -1068,6 +1068,8 @@ void DomainServer::processListRequestPacket(QSharedPointer<ReceivedMessage> mess
     // update the connecting hostname in case it has changed
     nodeData->setPlaceName(nodeRequestData.placeName);
 
+    nodeData->setLastDomainCheckinTimestamp(nodeRequestData.lastPingTimestamp);
+
```
Comment (roxanneskelly) [inline domain-server/src/DomainServer.cpp]: Add comment as to what lastPingTimestamp is

File: libraries/networking/src/NodeList.cpp [hunk 2]
```diff
@@ -618,12 +624,65 @@ void NodeList::processDomainServerConnectionTokenPacket(QSharedPointer<ReceivedM
...
+
+    quint64 domainServerRequestReceiveTime;
+    packetStream >> domainServerRequestReceiveTime;
+
+    quint64 domainServerPingSendTime;
+    packetStream >> domainServerPingSendTime;
+
+    qint64 pingLagTime = (now - qint64(connectRequestTimestamp)) / qint64(USECS_PER_MSEC);
+
+    qint64 domainServerRequestLag = (qint64(connectRequestTimestamp) - qint64(connectRequestTimestamp)) / qint64(USECS_PER_MSEC);
```
Comment (Atlante45) [inline libraries/networking/src/NodeList.cpp]: Typo, should be `domainServerRequestReceiveTime - connectRequestTimestamp`

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19654/HighFidelity-Beta-PR15626-c6de86a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19654/HighFidelity-Quest-Beta-PR15626-c6de86a.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19656/HighFidelity-Beta-PR15626-82f8687.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19656/HighFidelity-Quest-Beta-PR15626-82f8687.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19656/HighFidelity-Beta-PR15626-82f8687.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19656/HighFidelity-Beta-Interface-PR15626-82f8687.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19656/HighFidelity-Beta-PR15626-82f8687.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19656/HighFidelity-Beta-Interface-PR15626-82f8687.exe

**Pull request merged**